docs/content/querying/dimensionspecs.md

```diff
@@ -371,3 +371,29 @@ Following filtered dimension spec retains only the values matching regex. Note t
 ```
 
 For more details and examples, see [multi-valued dimensions](multi-valued-dimensions.html).
+
+### Lookup DimensionSpecs
+
+Lookup DimensionSpecs can be used to define directly a lookup implementation as dimension spec.
+Generally speaking there is two different kind of lookups implementations. 
+The first kind is passed at the query time like `map` implementation.
+
+```json
+{ 
+  "type":"lookup"
+  "dimension":"dimensionName"
+  "outputName":"dimensionOutputName"
+  "lookup":{"type": "map", "map":{"key":"value"}, "isOneToOne":false}
```

Comment:
@drcrallen we decided to move away from `LookupExtractionFn` in favor of implementing `DimensionSpec` and use the lookup delegator to do the apply/unapply. 
This has couple of advantages, lookups become less verbose, and optimaztion more easy to check for.
The fact that `LookupExtractor` exposes methods that are not included at the `ExtractionFn` API it doesn't make sense to use lookups via `ExtractionFN` API